doc/modules/grid_search.rst

```diff
@@ -1,10 +1,10 @@
-.. currentmodule:: sklearn.grid_search
+.. currentmodule:: sklearn.model_selection
 
 .. _grid_search:
 
-===============================================
-Grid Search: Searching for estimator parameters
-===============================================
+===========================================
+Tuning the hyper-parameters of an estimator
```

Comment:
I prefer "Tuning the hyper-parameters of an estimator" as it's more explicit.

The term "hyper-parameter" can be defined at the beginning of the section.